### DISCUSSION TITLE
DataCollectionOf as return type

### DISCUSSION THREAD

**Discussion by kmizzi** [spatie/laravel-data#628] [Ideas]
There doesn't seem to be a way to specify a DataCollectionOf as a return type.

Example:

```
public function getPricing(): #[DataCollectionOf(PricingData::class)] DataCollection
{
  return PricingData::collection([
     PricingData::from([...]),
     PricingData::from([...]),
  ]
}
```

This would return:
syntax error, unexpected token "#["

Any ideas?

**Comment by rubenvanassche** [comment]
That's not valid PHP indeed, use an attribute with @return to achieve such a thing.
